machine-learning/lambda-v2/lambda-artifacts/com.lambda.DLRObjectDetection/1.0.0/object_detection/prediction_utils.py

```diff
@@ -0,0 +1,286 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+import platform
+from ast import literal_eval
+from datetime import datetime
+from io import BytesIO
+from json import dumps
+from math import floor
+from os import path
+from sys import modules
+from time import sleep
+
+import config_utils
+import IPCUtils as ipc_utils
+from cv2 import (
+    COLOR_BGR2RGB,
+    COLOR_BGRA2RGB,
+    COLOR_GRAY2RGB,
+    FONT_HERSHEY_SIMPLEX,
+    IMWRITE_JPEG_QUALITY,
+    VideoCapture,
+    cvtColor,
+    destroyAllWindows,
+    imdecode,
+    imread,
+    imwrite,
+    putText,
+    rectangle,
+    resize,
+)
+from dlr import DLRModel
+from numpy import (
+    argsort,
+    array,
+    expand_dims,
+    float32,
+    frombuffer,
+    fromstring,
+    load,
+    transpose,
+    uint8,
+)
+
+config_utils.logger.info("Using dlr from '{}'.".format(modules[DLRModel.__module__].__file__))
+config_utils.logger.info("Using np from '{}'.".format(modules[argsort.__module__].__file__))
+config_utils.logger.info("Using cv2 from '{}'.".format(modules[VideoCapture.__module__].__file__))
+
+# Read labels file
+with open(config_utils.LABELS, "r") as f:
+    labels = literal_eval(f.read())
+
+dlr_model = DLRModel(config_utils.MODEL_DIR, config_utils.DEFAULT_ACCELERATOR)
+
+
+def transform_image(im):
+    if len(im.shape) == 2:
+        height, width = im.shape[:2]
+        im = expand_dims(im, axis=2)
+        nchannels = 1
+    elif len(im.shape) == 3:
+        height, width, nchannels = im.shape[:3]
+    else:
+        raise Exception("Unknown image structure")
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Using `Exception` and `BaseException` can make your code prone to errors and difficult to maintain. Instead, we recommend using one of the [Built-in Exceptions](https://docs.python.org/3/library/exceptions.html) or creating a custom exception class that is derived from `Exception` or one of its subclasses.